src/renderer/components/ft-list-lazy-wrapper/ft-list-lazy-wrapper.js

```diff
@@ -36,11 +36,33 @@ export default Vue.extend({
   computed: {
     hideLiveStreams: function() {
       return this.$store.getters.getHideLiveStreams
+    },
+    hideUpcomingPremieres: function () {
+      return this.$store.getters.getHideUpcomingPremieres
     }
   },
   methods: {
     onVisibilityChanged: function (visible) {
       this.visible = visible
+    },
+
+    showResult: function (data) {
+      if (!data.type) {
+        return false
+      }
+      if (data.type === 'video') {
+        if (this.hideLiveStreams && (data.liveNow || data.lengthSeconds == null)) {
+          // hide livestreams
+          return false
+        }
+        if (this.hideUpcomingPremieres &&
+            (data.isUpcoming || data.premiere || data.durationText === 'PREMIERE' ||
```

Comment:
I think we do need code comment to explain 
`data.premiere || data.durationText === 'PREMIERE'`

Why need both ...?

Reply:
Added comments explaining cases I've observed personally and removed cases I haven't observed.